### PR TITLE
feat: store guzzle PsrResponse

### PR DESCRIPTION
**Pull request by stixxx2k**
Store the PsrResponse for later use inside of the regular HttpResponse. All calls are backwards compatible

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "enm/json-api-client",
+  "name": "stixxx2k/json-api-client",
```
Comment (chris-doehring) [inline composer.json]: If I resolve this manually, your package name in the master branch will change to mine. Is that ok for you? If not, you have to create a new PR based on a branch where this matches my composer package name.

File: src/HttpClient/Response/HttpResponse.php [hunk 2]
```diff
@@ -28,12 +29,15 @@ class HttpResponse implements ResponseInterface
      */
     private $document;
 
+    /** @var PsrResponseInterface */
```
Comment (chris-doehring) [inline src/HttpClient/Response/HttpResponse.php]: This has to be nullable in the docs as well.

File: src/HttpClient/Response/HttpResponse.php [hunk 3]
```diff
@@ -28,12 +29,15 @@ class HttpResponse implements ResponseInterface
...
+    protected $psrResponse;
+
     /**
      * @param int $status
      * @param array $headers
      * @param DocumentInterface|null $document
      */
-    public function __construct(int $status, array $headers, ?DocumentInterface $document)
+    public function __construct(int $status, array $headers, ?DocumentInterface $document, ?PsrResponseInterface $psrResponse = null)
```
Comment (chris-doehring) [inline src/HttpClient/Response/HttpResponse.php:35]: Missing documentation above. You can also choose to remove them completely.

File: src/JsonApiClient.php [hunk 4]
```diff
@@ -154,7 +154,8 @@ public function execute(RequestInterface $request, bool $exceptionOnFatalError =
         $response = new HttpResponse(
             $httpResponse->getStatusCode(),
             $httpResponse->getHeaders(),
-            $this->createResponseBody($httpResponse->getBody()->getContents())
+            $this->createResponseBody($httpResponse->getBody()->getContents()),
+            $httpResponse
```
Comment (chris-doehring) [inline src/JsonApiClient.php:160]: What about calling `$httpResponse->getBody()->rewind()` before putting this instance into the client? This ensures the highest compatibility, as the body content still is present.

**Review by chris-doehring — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (stixxx2k) [reply composer.json]: Misinterpreted the PR mechanics. I've reverted the modification.

File: src/HttpClient/Response/HttpResponse.php (hunk 3 above)
Comment (stixxx2k) [reply src/HttpClient/Response/HttpResponse.php:35]: Removed the whole doc block

File: src/HttpClient/Response/HttpResponse.php (hunk 2 above)
Comment (stixxx2k) [reply src/HttpClient/Response/HttpResponse.php]: Yes

File: src/JsonApiClient.php (hunk 4 above)
Comment (stixxx2k) [reply src/JsonApiClient.php:160]: That's a valid objection.

**Review by chris-doehring — APPROVED** [review]
Thanks! Looks good now :)

**Comment by chris-doehring** [comment]
Your PR has been merged into master. I will release these changes as `v5.1` soon.

**Pull request merged**